tagstudio/src/qt/ts_qt.py

```diff
@@ -65,20 +65,38 @@
 QSettings.setPath(QSettings.IniFormat, QSettings.UserScope, os.getcwd())
 
 
-def open_file(path: str):
+def open_file(path: str, file_manager: bool = False):
 	try:
 		if sys.platform == "win32":
-			# Windows needs special attention to handle spaces in the file
-			# first parameter is for title, NOT filepath
-			subprocess.Popen(["start", "", os.path.normpath(path)], shell=True, close_fds=True, creationflags=subprocess.DETACHED_PROCESS)
+			normpath = os.path.normpath(path)
+			if file_manager:
+				command_name = "explorer"
+				command_args = [f"/select,{normpath}"]
+			else:
+				command_name = "start"
+				# first parameter is for title, NOT filepath
+				command_args = ["", normpath]
+			subprocess.Popen([command_name] + command_args, shell=True, close_fds=True, creationflags=subprocess.CREATE_NEW_PROCESS_GROUP | subprocess.CREATE_BREAKAWAY_FROM_JOB)
 		else:
 			if sys.platform == "darwin":
 				command_name = "open"
+				command_args = [path]
+				if file_manager:
+					# will reveal in Finder
+					command_args = command_args.append("-R")
```

Comment:
I tested on macOS and this was causing the command_args to become `None` due to the append function return breaking the string concatenation.

By doing this changed it works at intended


`# subprocess.Popen([command] + command_args, close_fds=True)
                     ^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: can only concatenate list (not "NoneType") to list`
 

```suggestion
					command_args.append("-R")
```

Reply:
Oh whoops, my LSP acts up in areas where it thinks code is impossible to run, which happens in these kind of blocks. Thank you for spotting this!